### ISSUE TITLE
[Request] Report conversation thread

### ISSUE THREAD

**Issue by tuningman9790** [HDInnovations/UNIT3D-Community-Edition#4354]
As a staff member when working on a report, you can only "write" something that resolves the report. But sometimes you'd like to ask questions to the person who did the reporting.

You'd then have to DM it personally and there is currently no way for other staffers to know what you have typed to this person on this specific subject.

Knowing this it would be interesting to be able to have a conversation in a report between person x & staff (staff are able to read & write/take over), claim a report and then click a button to resolve it.

**Comment by HDVinnie** [comment]
This is correct and honestly in the end the whole reports system need a ground up rewrite.

**Comment by Roardom** [comment]
> ground up rewrite.

I talked with a few people and came up with the following outline:

- [ ] Torrent reports are shown on the torrent page itself. The staff dashboard links to the torrent page.
- [ ] When a user reports a torrent, they can choose if the report should be private (i.e. only visible to staff) or visible to all users viewing the torrent page.
- [ ] Users can reply to reports visible to them. Replies to private reports will only be visible to staff and the original reporter.
- [ ] Torrents can have multiple reports, each with their own set of replies.
- [ ] Reports and replies can have the username remain anonymous or not
- [ ] The content of reports and replies can't be edited by the user (but the private and anon flags can be).
- [ ] Resolved reports are only visible to staff (hidden by default but a button can be clicked to show previous reports).
- [ ] BBCode support
